File: Onoicrm.Front/src/modules/treatment-plan/entities/TreatmentPlan.ts
import {Entity} from "~/shared/entities/Entity";

export class TreatmentPlan extends Entity{
  public patientId!:number
  public doctorId!:number
  public toothId!:number
  public clinicId!:number
  public bookingId!:number
  public caption!:string
  public description:string = ""

  constructor(init?:Partial<TreatmentPlan>) {
    super();
    Object.assign(this, init)
  }
}

File: Onoicrm.Domain/Entities/TreatmentPlan.cs
namespace Onoicrm.Domain.Entities;

public class TreatmentPlan : Entity
{
    public long ClinicId { get; set; }
    public Clinic? Clinic { get; set; }
    public long PatientId { get; set; }
    public UserProfile? Patient { get; set; }
    public long DoctorId { get; set; }
    public UserProfile? Doctor { get; set; }
    public DateTime DateTime { get; set; } = DateTime.Now.ToUniversalTime();
    public long ToothId { get; set; }
    public Tooth? Tooth { get; set; }
    public string Caption { get; set; } = "";
    public string Description { get; set; } = "";
    protected override int GetClassId() => ClassNames.TreatmentPlan;
}